### PR TITLE
python311Packages.succulent: init at 0.2.3

### PR DESCRIPTION
**Pull request by firefly-cpp**
###### Description of changes

Short description: _succulent_ is a pure Python framework that simplifies the configuration, management, collection, and preprocessing of data collected via POST requests.

[LINK](https://github.com/firefly-cpp/succulent)

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/succulent/default.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    flask
+    pandas
+    pyyaml
+  ];
+
+  # create flask and pandas versions consistent
+  preBuild = ''
+    toml-adapt -path pyproject.toml -a change -dep flask -ver X
+    toml-adapt -path pyproject.toml -a change -dep pandas -ver X
+  '';
```
Comment (fabaff) [inline pkgs/development/python-modules/succulent/default.nix:42]: IMHO, we should stick to `pythonRelaxDepsHook` when dealing with issues related to dependencies. `toml-adapt` seems to be able to do the job but my guess is that most of the reviewers are not familiar with it.

Comment (firefly-cpp) [reply pkgs/development/python-modules/succulent/default.nix]: Thanks. Fixed now.

**Comment by firefly-cpp** [comment]
@fabaff

**Comment by firefly-cpp** [comment]
@SuperSandro2000, please check this PR. @fabaff  is probably busy at the moment. We will need this package soon.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by firefly-cpp** [comment]
@SuperSandro2000, @wegank

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 240097` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.succulent</li>
    <li>python310Packages.succulent.dist</li>
    <li>python311Packages.succulent</li>
    <li>python311Packages.succulent.dist</li>
  </ul>
</details>

**Pull request merged**